File: src/main/java/mx/edu/unpa/adoptame/service/SolicitudAdopcionService.java
package mx.edu.unpa.adoptame.service;

import java.util.List;

import mx.edu.unpa.adoptame.domain.SolicitudAdopcion;

public interface SolicitudAdopcionService {

	SolicitudAdopcion crear(Integer idMascota, Integer idUsuarioSolicitante);

	List<SolicitudAdopcion> findRecibidasPendientes(Integer idUsuario);

	List<SolicitudAdopcion> findEnviadas(Integer idUsuario);

	SolicitudAdopcion aprobar(Integer idSolicitud, Integer idUsuarioOwner);
}

File: app/src/main/java/mx/edu/unpa/adoptame/service/SolicitudAdopcionService.kt
package mx.edu.unpa.adoptame.service

import mx.edu.unpa.adoptame.model.SolicitudAdopcion
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.PATCH
import retrofit2.http.POST
import retrofit2.http.Path

interface SolicitudAdopcionService {
    @POST("solicitudAdopcion")
    fun crear(@Body body: Map<String, @JvmSuppressWildcards Any>): Call<SolicitudAdopcion>

    @GET("solicitudAdopcion/recibidas/{idUsuario}")
    fun getRecibidas(@Path("idUsuario") idUsuario: Int): Call<List<SolicitudAdopcion>>

    @GET("solicitudAdopcion/enviadas/{idUsuario}")
    fun getEnviadas(@Path("idUsuario") idUsuario: Int): Call<List<SolicitudAdopcion>>

    @PATCH("solicitudAdopcion/{idSolicitud}/aprobar")
    fun aprobar(
        @Path("idSolicitud") idSolicitud: Int,
        @Body body: Map<String, @JvmSuppressWildcards Any>
    ): Call<SolicitudAdopcion>
}
